base/math.jl

```diff
@@ -72,6 +77,13 @@ function clamp!(x::AbstractArray, lo, hi)
     x
 end
 
+function clamp!(x::AbstractArray, ::Type{T}) where {T}
```

Comment:
I completely agree.  It seemed strange that the vector version of clamp has been deprecated but the in place version hadn't.